### PR TITLE
Remove driver serialized format as it is redundant

### PR DESCRIPTION
**Pull request by jatgarg**
This change removes the function `convertContainerToDriverSerializedFormat`.
This was used to convert the snapshot taken in detached container to a format which was understandable by driver for cases where app directly used the document service factory to create the container with the snapshot.
Now the snapshot returned by `container.serialize()` api is already in that format. So it reduces work done by apps.

### CODE REVIEW

File: packages/loader/container-loader/src/container.ts [hunk 1]
```diff
@@ -151,6 +151,34 @@ export enum ConnectionState {
     Connected,
 }
 
+// It specifies the summary taken by the serialize api on detached container.
+export interface IDetachedContainerSnapshot {
+    summary: ISummaryTree | ISnapshotTree,
+    // In format version >= 0.1, we would always have summary. However due to back-compat
+    // we would always have ISnapshotTree for version 0.0.
+    formatVersion: string,
```
Comment (jatgarg) [inline packages/loader/container-loader/src/container.ts:159]: Want to ask, can we avoid back compat here as I don't think anybody uses the serialize and rehydrate flow yet.

Comment (vladsud) [reply packages/loader/container-loader/src/container.ts:159]: That is correct, it's not used today. Though at some point we will bring it back

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -151,6 +151,34 @@ export enum ConnectionState {
...
+        const protocolSummaryTree = summaryTree.tree[".protocol"] as ISummaryTree;
+        const appSummaryTree = summaryTree.tree[".app"] as ISummaryTree;
+        assert(protocolSummaryTree !== undefined && appSummaryTree !== undefined,
+            "Protocol and App summary trees should be present");
+        snapshotTree = convertProtocolAndAppSummaryToSnapshotTree(
+            protocolSummaryTree,
+            appSummaryTree,
+        );
+    } else {
+        snapshotTree = detachedContainerSnapshot.summary as ISnapshotTree;
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:177]: There is no usage in old format, so you can likely remove it / replace to assert

File: packages/loader/container-loader/src/container.ts [hunk 3]
```diff
@@ -354,12 +382,23 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
...
-        await container.rehydrateDetachedFromSnapshot(snapshot);
+        const deserializedSummary = JSON.parse(snapshot);
+        let snapshotWithFormatVersion: IDetachedContainerSnapshot;
+        if (deserializedSummary.formatVersion === undefined) {
+            snapshotWithFormatVersion = {
+                summary: deserializedSummary,
+                formatVersion: "0.0",
+            };
+        } else {
+            snapshotWithFormatVersion = deserializedSummary;
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:398]: same here

File: packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts [hunk 4]
```diff
@@ -178,12 +181,12 @@ describeNoCompat(`Dehydrate Rehydrate Container Test`, (getTestObjectProvider) =
         it("Dehydrated container snapshot 2 times with changes in between", async () => {
             const { container, defaultDataStore } =
                 await createDetachedContainerAndGetRootDataStore();
-            const snapshotTree1: ISnapshotTree = JSON.parse(container.serialize());
+            const snapshotTree1: ISnapshotTree = convertDetachedContainerSnapshot(JSON.parse(container.serialize()));
```
Comment (vladsud) [inline packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts:184]: Looking at the usage, the name does not tell me really what it does.
getSnapsthotTreeFromSerializedContainer ?

**Review by vladsud — APPROVED** [review]

**Pull request merged**